jvm-packages/xgboost4j-spark/pom.xml

```diff
@@ -5,10 +5,11 @@
     <modelVersion>4.0.0</modelVersion>
     <parent>
         <groupId>ml.dmlc</groupId>
-        <artifactId>xgboost-jvm_2.12</artifactId>
+        <artifactId>xgboost-jvm</artifactId>
         <version>2.0.0-SNAPSHOT</version>
     </parent>
-    <artifactId>xgboost4j-spark_2.12</artifactId>
+    <name>xgboost4j-spark</name>
+    <artifactId>xgboost4j-spark_${scala.binary.version}</artifactId>
```

Comment:
It is safe for now. Sadly, making the script as spark adds more complexity than just using profiles. 
I've reused the approach discussed in the https://github.com/davidB/scala-maven-plugin/issues/177 
Spark approach originates in 2015 and maven changed since. I can make a script in place of maven profiles, if necessary.